packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -4823,6 +4823,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
                 return oldValue.text != newValue.text || oldValue.composing != newValue.composing;
               },
+              undoStackModifier: (TextEditingValue value) {
+                return defaultTargetPlatform == TargetPlatform.android ? value.copyWith(composing: TextRange.empty) : value;
```

Comment:
this is likely something that wants detailed documentation (for future readers and maybe future you)